feat(datasource): Add python-version datasource (#27583)
renovatebot/renovate: lib/modules/datasource/python-version/readme.md

> This datasource returns Python releases from [python.org](https://www.python.org/api/v2/downloads/release/) API.

Reviewer: ```suggestion
This datasource returns Python releases from the [python.org](https://www.python.org/api/v2/downloads/release/) API.
```

> This datasource returns Python releases from the [python.org](https://www.python.org/api/v2/downloads/release/) API.

Reviewer: ```suggestion
This datasource returns Python releases from the [python.org API](https://www.python.org/api/v2/downloads/release/).
```

Let's put the API text into the link, so it's clear where you'll end up after clicking the link.

> This datasource returns Python releases from the [python.org API](https://www.python.org/api/v2/downloads/release/).

Reviewer 2: Please also add info that it reads additional metadata from endoflife and github api

This datasource returns Python releases from the [python.org API](https://www.python.org/api/v2/downloads/release/).

It also fetches deprecated versions from [Endoflife Date Datasource](/modules/datasource/endoflife-date/)

> Because Renovate depends on [containerbase/python-prebuild](https://github.com/containerbase/python-prebuild/releases) it will also fetch Github API for releases there.

Reviewer: ```suggestion
Because Renovate depends on [`containerbase/python-prebuild`](https://github.com/containerbase/python-prebuild/releases) it will also fetch releases from the GitHub API.
```

## Example custom manager

> Below is a [custom regex manager](/modules/manager/regex/) for updating Python version in a Dockerfile. Python version in some cases omits dot separating major and minor number, ex.: `3.11` becomes `311`. So this example handles this case.

Reviewer: ```suggestion
Below is a [custom regex manager](/modules/manager/regex/) to update the Python versions in a Dockerfile.
Python versions sometimes drop the dot that separate the major and minor number: so `3.11` becomes `311`.
The example below handles this case.
```

This datasource returns Python releases from the [python.org API](https://www.python.org/api/v2/downloads/release/).

> It also fetches deprecated versions from [Endoflife Date Datasource](/modules/datasource/endoflife-date/)

Reviewer: ```suggestion
It also fetches deprecated versions from [Endoflife Date Datasource](/modules/datasource/endoflife-date/).
```

Sentences must end with a full stop.

Because Renovate depends on [`containerbase/python-prebuild`](https://github.com/containerbase/python-prebuild/releases) it will also fetch releases from the GitHub API.

## Example custom manager

Below is a [custom regex manager](/modules/manager/regex/) to update the Python versions in a Dockerfile.
Python versions sometimes drop the dot that separate the major and minor number: so `3.11` becomes `311`.
The example below handles this case.

```dockerfile
ARG PYTHON_VERSION=311
FROM image-python${PYTHON_VERSION}-builder:1.0.0
```

```json
"customManagers": [
  {
    "customType": "regex",
    "fileMatch": ["^Dockerfile$"],
    "matchStringsStrategy": "any",
    "matchStrings": ["ARG PYTHON_VERSION=\"?(?<currentValue>3(?<minor>\\d+))\"?\\s"],
    "autoReplaceStringTemplate": "ARG PYTHON_VERSION={{{replace '\\.' '' newValue}}}\n",
    "currentValueTemplate": "3.{{{minor}}}",
    "datasourceTemplate": "python-version",
    "versioningTemplate": "python",
    "depNameTemplate": "python"
  }
]

> ```

Reviewer: The linter is complaining about the JSON formatting:

> Error: Unexpected token : in JSON at position 16. Fix this manually by ensuring each block is a valid, complete JSON document.

This datasource returns Python releases from the [python.org API](https://www.python.org/api/v2/downloads/release/).

> It also fetches deprecated versions from the [Endoflife Date datasource](/modules/datasource/endoflife-date/).

Reviewer 2: ```suggestion
It also fetches deprecated versions from the [Endoflife Date datasource](../endoflife-date/index.md).
```
use relative paths, otherwise it will fail on mkdocs generation.

Because Renovate depends on [`containerbase/python-prebuild`](https://github.com/containerbase/python-prebuild/releases) it will also fetch releases from the GitHub API.

## Example custom manager

> Below is a [custom regex manager](/modules/manager/regex/) to update the Python versions in a Dockerfile.

Reviewer 2: ```suggestion
Below is a [custom regex manager](../../manager/regex/index.md) to update the Python versions in a Dockerfile.
```
